test/integration/scheduler/preemption_test.go

```diff
@@ -366,6 +367,47 @@ func TestDisablePreemption(t *testing.T) {
 	}
 }
 
+func TestPodPriorityResolution(t *testing.T) {
+	context := initTest(t, "podPriorityResolution")
+	defer cleanupTest(t, context)
+	cs := context.clientSet
+
+	tests := []struct{
+		PriorityClass string
+		Pod *v1.Pod
+	}{
+		{
+			PriorityClass: scheduling.SystemNodeCritical,
+			Pod: initPausePod(cs, &pausePodConfig{
+				Name:              fmt.Sprintf("pod1-%v", scheduling.SystemNodeCritical),
+				Namespace:         metav1.NamespaceSystem,
+				PriorityClassName: scheduling.SystemNodeCritical,
+			}),
+		},
+		{
+			PriorityClass: scheduling.SystemClusterCritical,
+			Pod: initPausePod(cs, &pausePodConfig{
+				Name:              fmt.Sprintf("pod2-%v", scheduling.SystemClusterCritical),
+				Namespace:         metav1.NamespaceSystem,
+				PriorityClassName: scheduling.SystemClusterCritical,
+			}),
+		},
+	}
+
+	pods := make([]*v1.Pod, len(tests))
+	for i, test := range tests {
+		pod, err := createPausePod(cs, test.Pod)
+		if err != nil {
+			t.Fatalf("Test [PodPriority/%v]: Error running pause pod: %v", test.PriorityClass, err)
+		}
+		pods[i] = pod
+		if pods[i].Spec.Priority == nil {
```

Comment:
Can you check the priority value as well?

Reply:
done